### PR TITLE
COP 1976: Data read table context

### PR DESCRIPTION
**Pull request by JimmySunDance**
## Description
This branch adds event context to the Data Read table, indicating to the user the success state of a document scan. 

## To test
- pull repo
- run `npm ci && npm run dev`
- If the electron window displays blank, refresh with `Cmd + r` or `Ctrl + r`
- To mock a document scan, press `F1`
- The "Open Chip" and "Read MRZ" columns should display the "Successful" tag. 

## Developer Checklist
\* Required

- [x] Up-to-date with master branch? \*

- [ ] Does it have tests?

- [x] Pull request URL linked to JIRA ticket? \*

- [ ] All reviewer comments resolved/answered? \*

### CODE REVIEW

File: packages/react/src/Components/Molecules/DataReadTable.jsx [hunk 1]
```diff
@@ -1,23 +1,48 @@
...
     <table className="govuk-table font--32">
       <caption className="govuk-table__caption">Data read</caption>
       <tbody className="govuk-table__body">
         <TableRow
-          rowLabel="Live to chip"
-          tagStatus="neutral"
-          tagText="No Data"
+          rowLabel="Open chip"
+          tagStatus={tagClassName(CD_SCDG1_CODELINE_DATA)}
+          tagText={tagText(CD_SCDG1_CODELINE_DATA)}
```
Comment (Ali-Baker) [inline packages/react/src/Components/Molecules/DataReadTable.jsx:40]: `tagText` takes 2 arguments.

File: packages/react/src/Components/Molecules/DataReadTable.jsx [hunk 2]
```diff
@@ -1,23 +1,48 @@
 // Global imports
-import React from 'react';
+import React, { useContext } from 'react';
 
 // Local imports
 import TableRow from './TableRow';
+import { EventSourceContext } from '../Context/EventSource';
+
+const tagText = (data, chipData) => {
```
Comment (Ali-Baker) [inline packages/react/src/Components/Molecules/DataReadTable.jsx:8]:
```suggestion
const tagText = (codelineData, chipData) => {
```

Comment (Ali-Baker) [reply packages/react/src/Components/Molecules/DataReadTable.jsx:8]: or `imageData` ?

File: packages/react/src/Components/Molecules/DataReadTable.jsx [hunk 3]
```diff
@@ -1,23 +1,48 @@
...
+  if (chipData) {
+    if (data.data === chipData.data) {
+      return 'successful';
+    }
+    return 'warning';
+  }
+  return data ? 'successful' : 'No data';
```
Comment (Ali-Baker) [inline packages/react/src/Components/Molecules/DataReadTable.jsx:15]:
```suggestion
  if (data.data === chipData?.data || data) {
    return 'successful';
  } elseif (data.data !== chipData?.data) {
    return 'warning';
  }
  return 'No data';
```

Comment (Ali-Baker) [reply packages/react/src/Components/Molecules/DataReadTable.jsx:15]: Could that be better to read?

It took me sometime to know what this function is doing.

File: packages/react/src/Components/Molecules/DataReadTable.jsx (hunk 1 above)
Comment (JimmySunDance) [reply packages/react/src/Components/Molecules/DataReadTable.jsx:40]: For this tag, we only need to check if the Chip was opened. The second argument has now been given a default value of null.

File: packages/react/src/Components/Molecules/DataReadTable.jsx (hunk 2 above)
Comment (JimmySunDance) [reply packages/react/src/Components/Molecules/DataReadTable.jsx:8]: Changed to `primaryTagData` , the first argument can be chip data or scan data, second argument will always be chip data.

File: packages/react/src/Components/Molecules/DataReadTable.jsx (hunk 1 above)
Comment (Ali-Baker) [reply packages/react/src/Components/Molecules/DataReadTable.jsx:40]: 👍

File: packages/react/src/Components/Molecules/DataReadTable.jsx [hunk 4]
```diff
@@ -1,23 +1,48 @@
 // Global imports
-import React from 'react';
+import React, { useContext } from 'react';
```
Comment (Ali-Baker) [inline packages/react/src/Components/Molecules/DataReadTable.jsx:2]: Reads better?
```suggestion
  const noData = !chipData && !primaryTagData;
  if (noData) return 'No data';

  const isPrimaryTagEqualChip =
    chipData && primaryTagData.data && chipData.data;
  return isPrimaryTagEqualChip ? 'successful' : 'warning';
```

Comment (JimmySunDance) [reply packages/react/src/Components/Molecules/DataReadTable.jsx:2]: As you pointed out, the base logic isnt quite right so I have made some light adjustments.

**Pull request closed without merging**